docs/cpp/source/notes/tensor_cuda_stream.rst

```diff
@@ -0,0 +1,277 @@
+
+  // create a tensor on device 0
+  torch::Tensor tensor0 = torch::ones({2, 2}, torch.device(torch::kCUDA));
+  // get a new CUDA stream from CUDA stream pool on device 0 
+  at::cuda::CUDAStream myStream = at::cuda::getStreamFromPool();
+  // set current CUDA stream from default stream to `myStream` on device 0
+  at::cuda::setCurrentCUDAStream(myStream);
+  // sum() on tensor0 uses `myStream` as current CUDA stream
+  tensor0.sum();
+
+  // get the default CUDA stream on device 0 
+  at::cuda::CUDAStream defaultStream = at::cuda::getDefaultCUDAStream();
+  // set current CUDA stream back to default CUDA stream on devide 0
+  at::cuda::setCurrentCUDAStream(defaultStream);
+  // sum() on tensor0 uses `defaultStream` as current CUDA stream
+  tensor0.sum();
+
+.. code-block:: cpp
+
+  // This example is the same as previous example, but explicitly specify device
+  // index and use CUDA stream guard to set current CUDA stream
+
+  // create a tensor on device 0
+  torch::Tensor tensor0 = torch::ones({2, 2}, torch.device(torch::kCUDA));
+  // get a new stream from CUDA stream pool on device 0
+  at::cuda::CUDAStream myStream = at::cuda::getStreamFromPool(false, 0);
+  // set the current CUDA stream to `myStream` within the bracket scope using CUDA stream guard
+  {
+    at::cuda::CUDAStreamGuard guard(myStream);
+    // current CUDA stream is `myStream` from here till the end of bracket. 
+    // sum() on tensor0 uses `myStream` as current CUDA stream
+    tensor0.sum();
+  }
+  // current CUDA stream is reset to default CUDA stream after CUDA stream guard is destroyed
+  // sum() on tensor0 uses default CUDA stream on device 0 as current CUDA stream
+  tensor0.sum();
+
+.. attention::
+  
+  Above code is running on the same CUDA device. `setCurrentCUDAStream` will always set current CUDA stream on current device, but please
+  pay attention on that `setCurrentCUDASteram` actually set current stream on the device of passed in CUDA stream.
+  
+
+2. Acquiring and setting CUDA stream on multi-devices.
+
+.. code-block:: cpp
+
+  // This example shows how to acquire and set CUDA stream on two devices.
+
+  // acquire new CUDA streams from CUDA stream pool on device 0 and device 1
+  at::cuda::CUDAStream myStream0 = at::cuda::getStreamFromPool(false, 0);
+  at::cuda::CUDAStream myStream1 = at::cuda::getStreamFromPool(false, 1);
+
+  // set current CUDA stream to `myStream0` on device 0
+  at::cuda::setCurrentCUDAStream(myStream0);
+  // set current CUDA stream to `myStream1` on device 1
+  at::cuda::setCurrentCUDAStream(myStream1);
+
+  // create a tensor on device 0, no need to specify device index since 
+  // current device index is 0
+  torch::Tensor tensor0 = torch::ones({2, 2}, torch::device(at::kCUDA));
+  // sum() on tensor0 use `myStream0` as current CUDA stream on device 0
+  tensor0.sum();
+
+  // change the current device index to 1 by using CUDA device guard within a braket scope
```

Comment:
"braket" -> "curly brace" OR you can just remove the word and say "... within a scope"